### DISCUSSION TITLE
technical-writings/2024/10/21/openais-swarm/

### DISCUSSION THREAD

**Discussion by giscus** [pampa-labs/pampa-labs.github.io#4] [Q&A]
# technical-writings/2024/10/21/openais-swarm/

Insights and solutions for LLM technologies by Pampa Labs.

https://pampa-labs.github.io/technical-writings/2024/10/21/openais-swarm/

**Comment by lgesuellip** [comment]
Defining tools strictly as Python functions is a limitation, making the framework less flexible. It should allow more flexible formats, such as JSON format, similar to defining tools from scratch with the OpenAI client. 
I adapted the framework to support this approach [here](https://github.com/lgesuellip/swarm/tree/feature/add_json_format), just for testing purposes.
